contracts/BitGuildMarketplace.sol

```diff
@@ -0,0 +1,355 @@
+pragma solidity ^0.4.22;
+
+import "./BitGuildToken.sol";
+import "./BitGuildAccessAdmin.sol";
+import "./BitGuildWhitelist.sol";
+import "./BitGuildFeeOracle.sol";
+import "./SafeMath.sol";
+
+
+// @dev Interface for ERC-721 NFT
+contract ERC721Interface {
+    event Transfer(address indexed _from, address indexed _to, uint256 _tokenId);
+    event Approval(address indexed _owner, address indexed _approved, uint256 _tokenId);
+    event ApprovalForAll(address indexed _owner, address indexed _operator, bool _approved);
+    function balanceOf(address _owner) external view returns (uint256);
+    function ownerOf(uint256 _tokenId) external view returns (address);
+    function safeTransferFrom(address _from, address _to, uint256 _tokenId, bytes data) external;
+    function safeTransferFrom(address _from, address _to, uint256 _tokenId) external;
+    function transferFrom(address _from, address _to, uint256 _tokenId) external;
+    function approve(address _approved, uint256 _tokenId) external;
+    function setApprovalForAll(address _operator, bool _approved) external;
+    function getApproved(uint256 _tokenId) external view returns (address);
+    function isApprovedForAll(address _owner, address _operator) external view returns (bool);
+}
+
+/// @title ERC-721 Non-Fungible Token Standard
+interface ERC721TokenReceiver {
+	function onERC721Received(address _from, uint256 _tokenId, bytes data) external returns(bytes4);
+	function onERC721Received(address _operator, address _from, uint256 _tokenId, bytes data) external returns(bytes4);
+}
+
+/*
+ * @title BitGuildMarketplace
+ * @dev: Marketplace smart contract for BitGuild.com
+ */
+contract BitGuildMarketplace is BitGuildAccessAdmin, BitGuildWhitelist {
+    using SafeMath for uint256;
+
+    //from zepellin ERC721Receiver.sol
+    //old version
+    bytes4 constant ERC721_RECEIVED_OLD = 0xf0b9e5ba;
+    //new version with operator
+    bytes4 constant ERC721_RECEIVED = 0x150b7a02;
+
+    // Predefined PLAT token
+    // BitGuildToken public token = BitGuildToken(0x7E43581b19ab509BCF9397a2eFd1ab10233f27dE); // Main Net
+    BitGuildToken public PLAT = BitGuildToken(0x0F2698b7605fE937933538387b3d6Fec9211477d); // Rinkeby
+
+    uint public defaultExpiry = 7 days;   // default expiry 7 days
+    // TODO: add function to modify this length 
+
+    uint public currentItemId = 1; // auto-increment
+
+    mapping(uint256 => address) public itemIdToContract;       // game contract the item belongs to
+    mapping(uint256 => address) public itemIdToSeller;         // seller of the item. 0x0 if sold/delisted
+    mapping(uint256 => uint256) public itemIdToTokenId;        // tokenId for this item
+    mapping(uint256 => uint256) public itemIdToPLAT;           // price in PLAT for this item
+    mapping(uint256 => uint256) public itemIdToETH;            // price in ETH for this item
+    mapping(uint256 => uint256) public itemIdToCreatedAt;      // timestamp when item is listed
```

Comment:
All of that basically needs to be replaced with itemHash to orderData mapping. With the latter being a structure with price, currency enum, createdAt.